KeyVault.Acmebot/Providers/DnsMadeEasyProvider.cs

```diff
@@ -0,0 +1,210 @@
+        }
+
+        public async Task DeleteTxtRecordAsync(DnsZone zone, string relativeRecordName)
+        {
+            var records = await _client.ListRecordsAsync(zone.Id);
+
+            var recordsToDelete = records.Where(r => r.Name == relativeRecordName && r.Type == "TXT");
+
+            foreach (var record in recordsToDelete)
+            {
+                await _client.DeleteRecordAsync(zone.Id, record);
+            }
+        }
+
+        private class DnsMadeEasyClient
+        {
+            public DnsMadeEasyClient(string apiKey, string secretKey)
+            {
+                _httpClient = new HttpClient(new ApiKeyHandler(apiKey, secretKey, new HttpClientHandler()))
+                {
+                    BaseAddress = new Uri("https://api.dnsmadeeasy.com/V2.0/")
+                };
+
+                _httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            }
+
+            private readonly HttpClient _httpClient;
+
+            public async Task<IReadOnlyList<Domain>> ListZonesAsync()
+            {
+                var response = await _httpClient.GetAsync("dns/managed");
+
+                response.EnsureSuccessStatusCode();
+
+                var domains = await response.Content.ReadAsAsync<ListDomainsResult>();
+
+                return domains.Domains;
+            }
+
+            public async Task<IReadOnlyList<DnsEntry>> ListRecordsAsync(string zoneId)
+            {
+                var response = await _httpClient.GetAsync($"dns/managed/{zoneId}/records");
+
+                response.EnsureSuccessStatusCode();
+
+                var entries = await response.Content.ReadAsAsync<ListDnsEntriesResponse>();
+
+                return entries.DnsEntries;
+            }
+
+            public async Task DeleteRecordAsync(string zoneId, DnsEntry entry)
+            {
+                var response = await _httpClient.SendAsync(new HttpRequestMessage(HttpMethod.Delete, $"dns/managed/{zoneId}/records/{entry.Id}"));
+
+                response.EnsureSuccessStatusCode();
+            }
+
+            public async Task AddRecordAsync(string zoneId, DnsEntry entry)
+            {
+                var response = await _httpClient.SendAsync(new HttpRequestMessage(HttpMethod.Post, $"dns/managed/{zoneId}/records")
+                {
+                    Content = new StringContent(JsonConvert.SerializeObject(entry), Encoding.UTF8, "application/json")
+                });
+
+                response.EnsureSuccessStatusCode();
+            }
+
+            internal sealed class ApiKeyHandler : DelegatingHandler
+            {
+                private string ApiKey { get; }
+                private HMACSHA1 HMAC { get; }
+
+                public ApiKeyHandler(string apiKey, string secretKey, HttpMessageHandler innerHandler) : base(innerHandler)
+                {
+                    if (apiKey is null)
+                        throw new ArgumentNullException(nameof(apiKey));
```

Comment:
All if statements must have an explicit scope. This may be redundant, but it is a coding convention.

```suggestion
                    if (apiKey is null)
                    {
                        throw new ArgumentNullException(nameof(apiKey));
                    }
```